### ISSUE TITLE
No output from fisher.py

### ISSUE THREAD

**Issue by pyongheng** [TheBrownLab/PhyloFisher#29]
I am still unable to run informant.py
Looking through the files, there was no output from fisher.py aside from original_names.tsv. 
fisher.py reports a number of genes were excluded due to no hmm hits (96 genes) but ends with Fisher completed without any errors.

Looking through the temp files, both diamond.res are empty files. 
For some genes the .hmmout file do have hits. 

I get the same result when using --all_bbh
Using Version: 1.1.2

**Comment by atice** [comment]
Hi @pyongheng,

Could you please provide us with the following so we can better help trouble shoot your issue?

1. Your OS and version
2. The exact command you ran to produce the error and all commands run beforehand
3. Your input_metadata.tsv file.
4. Your config_ini file
5. Can you please tell us if you are using our provided starting database or a custom database you created.

Thanks
Alex

**Comment by pyongheng** [comment]
1. macOS Monterey v12.2.1
2. commands ran
`conda activate fisher
explore_database.py -d /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database -h
explore_database.py -d /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database -t
explore_database.py -d /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database -l
fisher.py
informant.py -i fisher_out_Jul.27.2022 `
this produces 

> Traceback (most recent call last):
  File "/Users/yonghengphua/opt/anaconda3/envs/fisher/bin/informant.py", line 278, in <module>
    res = table_with_routes(matrix, routes)
  File "/Users/yonghengphua/opt/anaconda3/envs/fisher/bin/informant.py", line 101, in table_with_routes
    df.loc[taxon] = len(df.columns) * [0]
  File "/Users/yonghengphua/opt/anaconda3/envs/fisher/lib/python3.7/site-packages/pandas/core/indexing.py", line 692, in __setitem__
    iloc._setitem_with_indexer(indexer, value, self.name)
  File "/Users/yonghengphua/opt/anaconda3/envs/fisher/lib/python3.7/site-packages/pandas/core/indexing.py", line 1629, in _setitem_with_indexer
    self._setitem_with_indexer_missing(indexer, value)
  File "/Users/yonghengphua/opt/anaconda3/envs/fisher/lib/python3.7/site-packages/pandas/core/indexing.py", line 1903, in _setitem_with_indexer_missing
    raise ValueError("cannot set a frame with no defined columns")
ValueError: cannot set a frame with no defined columns

3.  input file

> Location	File Name	Unique ID	Higher Taxonomy	Lower Taxonomy	Blast Seed	Long Name	Data Type	Source
.	transcripts.fasta.transdecoder.pep	Haplo	Alveolata	Dinoflagellata	none	Haplozoon sp.	transcriptomic	Haplo_SmartSeq

4. config file

> database_folder = /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database/
input_file = input_metadata.tsv
orthomcl = /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database/orthomcl
color_conf = /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database/tree_colors.tsv

5. I'm using the starting provided database.

**Comment by robert-ervin-jones** [comment]
Hi @pyongheng,

A couple of questions:

1. Are you using an Intel or M1 Mac?
2. Did you run the commands exactly as shown? I'm asking because I don't believe you did since they are all together on one line. If you didn't run precisely what is shown, can you share the commands you used with a newline separating them? 
3. Can you confirm `transcripts.fasta.transdecoder.pep` is in FASTA format?

Thanks,
Robert

**Comment by pyongheng** [comment]
Hi Robert,

Here are the details

1. M1 Mac
2. The markdown I used was wrong in the previous comment.
This is with the config.ini already in the folder
```
conda activate fisher 
explore_database.py -d /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database -h 
explore_database.py -d /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database -t 
explore_database.py -d /Users/yonghengphua/Documents/programmes/database/PhyloFisherDatabase_v1.0/database -l 
fisher.py 
informant.py -i fisher_out_Jul.27.2022
```

4. transcripts.fasta.transdecoder.pep is in FASTA format

**Comment by atice** [comment]
Hi @pyongheng,

If your input_metadata.tsv is structured as in your above comment then this is most likely the problem. You have spaces rather than tabs separating the information in this file. Can you check and if they are indeed spaces, change then to tabs and run fisher.py again.

Alex

**Comment by atice** [comment]
Hi @pyongheng, 

One additional comment, are you using the intel version of anaconda or the M1 version? There are known issues with some of the third party tools used by fisher.py when the M1 version is used. If you are using the M1 version please uninstall teh M1 version and install the intel version which runs fine on M1 Macs.

Alex

**Comment by pyongheng** [comment]
Hi Alex, 

For the input_metadata.tsv they are separated using tabs. 
I checked them and re-ran fisher.py and still got the same result. 
Input seems fine to me

- in temp files, for_diamond.fasta has sequences in it 
- both dataset_diamond.res and orthomcl_diamond.res are empty

I am unsure if I am using the intel version of anaconda but I am not using the M1 version.  
From anaconda download page I got it using 64-Bit Graphical Installer, the first option.  
I reinstalled it but still got the same issue.
